src/main/java/software/amazon/awssdk/crt/auth/signing/AwsSigner.java

```diff
@@ -25,7 +26,10 @@
         CompletableFuture<HttpRequest> future = new CompletableFuture<HttpRequest>();
 
         try {
-            awsSignerSignRequest(request, config, future);
+            //JNI is much easier to deal with primitives, so go ahead and hoist the headers into an array.
+            HttpHeader[] tempHeadersToSign = new HttpHeader[request.getHeaders().size()];
+            tempHeadersToSign = request.getHeaders().toArray(tempHeadersToSign);
+            awsSignerSignRequest(request, tempHeadersToSign, config, future);
```

Comment:
trivial: worth adding a HttpRequest.getHeadersArray() or a helper function somewhere?

Reply:
i can do that